### PR TITLE
[SPARK-7646][SQL] Create table from dataframe support for JDBCDataSource

### PR DESCRIPTION
**Pull request by gvramana**
Supported  CreatableRelationProvider support for JDBCDataSource. 
df.saveAsTable to JDBC data source, should create a table and write dataframe content to table.

### CODE REVIEW

**Comment by rxin** [comment]
Jenkins, test this please.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/MetastoreDataSourcesSuite.scala [hunk 1]
```diff
@@ -18,11 +18,14 @@
 package org.apache.spark.sql.hive
 
 import java.io.File
+import java.sql.DriverManager
```
Comment (viirya) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/MetastoreDataSourcesSuite.scala]: Why not put these tests into `JDBCSuite` but `MetastoreDataSourcesSuite`?

Comment (gvramana) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/MetastoreDataSourcesSuite.scala]: Currently SaveAsTable is supported in Hive context only, Also SchemaRelationProvider interface is impletemented along with CreatableRelationProvider.

**Comment by rxin** [comment]
I'm going to close this pull request. If this is still relevant and you are interested in pushing it forward, please open a new pull request. Thanks!

**Pull request closed without merging**